### DISCUSSION TITLE
Can custom logger extending `LightningLoggerBase` be imported from separate library?

### DISCUSSION THREAD

**Discussion by shnela** [Lightning-AI/pytorch-lightning#6787] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I'm responsible for implementing `pytorch_lightning.loggers.neptune.NeptuneLogger`.

Right now we're going through massive api update for neptune client, which requires updating `NeptuneLogger` in your repository (`PyTorchLightning/pytorch-lightning`).

Theoretically we could implement our `class NeptuneLogger(LightningLoggerBase)` in our repo, and import this class as module in `PyTorchLightning/pytorch-lightning`.
In such configuration we could easily update logger in our repo without having to create PR in `PyTorchLightning/pytorch-lightning`, when updating our api again in the future.

Are there any reasons against such solution?

**Comment by awaelchli** [comment]
Hi 

Thanks for your interest and notification about the upcoming changes.
It would be fine with us if the NeptuneLogger moves to your repo and we would certainly appreciate if the Neptune developers can keep it updated. I think it's a great idea, also because if the user updates Neptune they get the new Logger updates automatically and don't have to wait for a Lightning update :)

**Reply by shnela** [reply]
Indeed such solution will make updates of [neptune-client](https://github.com/neptune-ai/neptune-client) much easier and reliable,  but there's one more thing to take into consideration.

What if you'd like to change your pytorch-lightning logger API?
How likely do you think, changes to `BaseLogger`, which force immediate changes in inherited logger code, may occur?

**Reply by awaelchli** [reply]
can't say for sure but I think not so likely. The logger base class has been quite stable and has not seen many breaking api changes as far as I can see. There were mostly only additions of features that are optional.

**Comment by edenlightning** [comment, marked as answer]
Go for it! This is the best way to make sure users get the best experience.

**Reply by shnela** [reply]
Hi @edenlightning  , we've decided to try this solution, but first we want to simply update our loggers in your repo one more time.

Could you please take a look at our PR #6867 ?
It's marked as Draft, but last thing to do is simply documentation update.
